dashboard/context_processors.py

```diff
@@ -1,14 +1,15 @@
-from django.db import connection as conn
-from django.conf import settings
+from dashboard.common.util import *
 
 def course_name(request):
-    course_id = str(settings.UDW_ID_PREFIX) + str(request.resolver_match.kwargs.get('course_id'))
+    course_id = str(request.resolver_match.kwargs.get('course_id'))
     course_name = "Course Not Found"
     if (course_id): 
-        cursor = conn.cursor()
-        cursor.execute("SELECT name FROM course WHERE id = %s", [course_id])
-        row = cursor.fetchone()
-        if (row != None):
-            course_name = row[0]
+        course_name = get_course_name_from_id(course_id)
 
     return {'course_name': course_name}
+
+def current_user_course_id(request):
+    course_id = str(request.resolver_match.kwargs.get('course_id'))
+    if not course_id:
+        course_id = get_default_user_course_id(request.user.username)
```

Comment:
I would add a `log.info` statement capturing this case

Reply:
Yeah, this is going to be called many times in a single request so there'd be many duplicate records. I wonder if there's a way in python to buffer/compress log messages like for instance if I had.

logger.info("Course ID not in request for someuser")

And that was repeated like 6 times in the same requests it would just say

Course ID not in request for some user (Message repeated 6 times)

Reply:
That many time it would be noise, But why is it calling 6 times in one request. I am assuming this is from the home page where course id is required for each view. But once you get it from the backend we should store it as JS variable and use it. I don't want to make it complicated if it is practical enough to do it. Since the `current_user_course_id ` is not making a DB call we should be fine for calling it multiple time

Reply:
Well this function is being used currently by a context_processor and on every page it's called 7 times. But you make a great point. These functions should for sure be cached or memoized on the backend and just called once by the view.

Reply:
I think caching could really improve this app, especially the views. This probably won't be a worry for the pilot but something we really should look at for a larger rollout.

https://docs.djangoproject.com/en/1.11/topics/cache/

Reply:
I added that log to the method, it looks like even though it's hit multiple times in the template it's only executed once.